dotnet/src/Functions/Functions.OpenApi/OpenApi/OpenApiDocumentParser.cs

```diff
@@ -229,6 +234,127 @@ private static List<RestApiOperationServer> CreateRestApiOperationServers(IList<
+            BearerFormat = securityScheme.BearerFormat,
+            Flows = CreateRestApiOAuthFlows(securityScheme.Flows),
+            OpenIdConnectUrl = securityScheme.OpenIdConnectUrl
+        };
+    }
+
+    /// <summary>
+    /// Build a <see cref="RestApiOAuthFlows"/> object from the given <see cref="OpenApiOAuthFlows"/> object.
+    /// </summary>
+    /// <param name="flows">The REST API OAuth flows.</param>
+    private static RestApiOAuthFlows? CreateRestApiOAuthFlows(OpenApiOAuthFlows? flows)
+    {
+        return flows is not null ? new RestApiOAuthFlows()
+        {
+            Implicit = CreateRestApiOAuthFlow(flows.Implicit),
+            Password = CreateRestApiOAuthFlow(flows.Password),
+            ClientCredentials = CreateRestApiOAuthFlow(flows.ClientCredentials),
+            AuthorizationCode = CreateRestApiOAuthFlow(flows.AuthorizationCode),
+        } : null;
+    }
+
+    /// <summary>
+    /// Build a <see cref="RestApiOAuthFlow"/> object from the given <see cref="OpenApiOAuthFlow"/> object.
+    /// </summary>
+    /// <param name="flow">The REST API OAuth flow.</param>
+    private static RestApiOAuthFlow? CreateRestApiOAuthFlow(OpenApiOAuthFlow? flow)
+    {
+        return flow is not null ? new RestApiOAuthFlow()
+        {
+            AuthorizationUrl = flow.AuthorizationUrl,
+            TokenUrl = flow.TokenUrl,
+            RefreshUrl = flow.RefreshUrl,
+            Scopes = new ReadOnlyDictionary<string, string>(flow.Scopes ?? new Dictionary<string, string>())
+        } : null;
+    }
+
+    /// <summary>
+    /// Build a list of <see cref="RestApiSecurityRequirement"/> objects from the given <see cref="OpenApiSecurityRequirement"/> objects.
+    /// </summary>
+    /// <param name="security">The REST API operation security</param>
+    /// <param name="globalRequirements">Existing global security requirements</param>
+    private static List<RestApiSecurityRequirement> CreateRestApiOperationSecurityRequirements(IList<OpenApiSecurityRequirement>? security, List<RestApiSecurityRequirement>? globalRequirements = null)
+    {
+        var operationRequirements = new List<RestApiSecurityRequirement>();
+
+        if (security is not null)
+        {
+            foreach (var item in security)
+            {
+                foreach (var keyValuePair in item)
+                {
+                    if (keyValuePair.Key is not OpenApiSecurityScheme openApiSecurityScheme)
+                    {
+                        throw new KernelException("The security scheme is not supported.");
+                    }
+
+                    operationRequirements.Add(new RestApiSecurityRequirement(new Dictionary<RestApiSecurityScheme, IList<string>> { { CreateRestApiSecurityScheme(openApiSecurityScheme), keyValuePair.Value } }));
+                }
+            }
+        }
+
+        // add globally defined security requirements that are not overridden at the operation level
+        if (globalRequirements is not null)
```

Comment:
I wonder whether we need to propagate global security requirements down to operations. Alternatively, we can keep the requirements at the global level, have a new property for that in the `RestApiSpecification` class, and leave the inheritance/propagation functionality to the consumers. This way, we will provide more granular information to the consumers.